pkg/traceability/sampling/sample.go

```diff
@@ -20,9 +21,28 @@ func (s *sample) ShouldSampleTransaction(details TransactionDetails) bool {
 	if hasFailedStatus && s.config.ReportAllErrors {
 		return true
 	}
-	if s.config.PerAPI && details.APIID != "" {
-		return s.shouldSampleWithCounter(details.APIID)
+
+	if s.config.PerAPI && s.config.PerSub {
+		if details.SubID != "" {
+			return s.shouldSampleWithCounter(fmt.Sprintf("%s-%s", details.APIID, details.SubID))
+		}
+		if details.APIID != "" {
+			return s.shouldSampleWithCounter(details.APIID)
+		}
+	}
```

Comment:
I was thinking, in this case where both settings are on.

We probably want to count every apiid, even if subid and apiid are both set.  That way when subid is not set we still have the counts of all transactions for that apiid.

Reply:
Ok, what should change here?

Reply:
Something like this, maybe?

```
if s.config.PerAPI && s.config.PerSub {
	if details.SubID != "" {
		s.shouldSampleWithCounter(details.APIID). // keep count of sampling by just api id as well
		return s.shouldSampleWithCounter(fmt.Sprintf("%s-%s", details.APIID, details.SubID))
	}
	if details.APIID != "" {
		return s.shouldSampleWithCounter(details.APIID)
	}
}
```